docs/freebsd/README.md

```diff
@@ -61,12 +61,36 @@ autoboot_delay="-1"
 console="comconsole"
 __EOF__
 ```
-Fetch a copy of the FreeBSD kernel sources and place them in `/usr/src`.  For instance, to get a copy of the current development sources, run:
-
+Fetch a copy of the FreeBSD kernel sources and place them in `/usr/src`. You will likely need to expand the disk of the VM. Make sure the VM is off. On the host:
+```console
+# truncate -s 15G $IMAGEFILE
+```
+After booting VM back up, run:
+```console
+# /etc/rc.d/growfs onestart
+```
+To get a copy of the current development sources, run:
 ```console
 # pkg install git
 # git clone --depth=1 --branch=master https://github.com/freebsd/freebsd /usr/src
 ```
+If your VM does not connect to the Internet, you can instead checkout the kernel sources from the host. Make sure the VM is off. On the host:
+```console
+# mdconfig -a -f $IMAGEFILE
+# mount /dev/md0p4 /mnt
+# cp /etc/resolv.conf /mnt/etc
+# mount -t devfs none /mnt/dev
+# chroot /mnt /bin/sh
+```
+This will give you a shell in the VM image where you can install packages. Before booting the VM again, make sure to run:
+```
+# umount /mnt/dev
+# umount /mnt
+```
+The md device will linger and you can use it again later if you want. Otherwise destroy it:
+```
+# mdconfig -d -u 0
+```
```

Comment:
All of this makes me think that it's better to always do the kernel build on the host. In general you probably don't want your syzkaller VMs having any external network access anyway.

So, one would download the sources onto the host, add a SYZKALLER config file as described below, run buildkernel, and then run installkernel with DESTDIR=/mnt.